### PR TITLE
Application API endpoint views and methods

### PR DESCRIPTION
**Pull request by nyashabryan**

### CODE REVIEW

File: .gitignore
Comment (KaitCrawford) [inline .gitignore]: Please add a new line at the end of this file. It's a good convention to follow because some systems use and empty line to indicate where then end of a file is when they are processing it.

**Review by sewagodimo — APPROVED** [review]
looks good to me :cake:

File: cspatients/util.py
Comment (KaitCrawford) [inline cspatients/util.py]: Please move this to just below the `json.loads(data)` line
It's advisable to keep only the lines relevant to the try inside the try block.
There are a number of things currently inside your try block that could through all sorts of errors that are unrelated to the json decoding.

File: cspatients/util.py
Comment (KaitCrawford) [inline cspatients/util.py]: Please add a comment explaining why 'All Responses' needs to be treated differently here

File: cspatients/util.py
Comment (KaitCrawford) [inline cspatients/util.py]: Just nit-picking here: please rename this variable to `fields`

File: cspatients/util.py
Comment (KaitCrawford) [inline cspatients/util.py]: As before, please move this `except` closer to the `try` so that the try block only contains the relevant code.

File: cspatients/util.py
Comment (KaitCrawford) [inline cspatients/util.py]: I'm not sure why this except is necessary since you're not trying to retrieve a Patient anywhere?
Is it loaded in one of the methods called in the block? If so, which one?

File: cspatients/util.py
Comment (KaitCrawford) [inline cspatients/util.py]: This line is exactly the same as the one in the if statement above, is it necessary to have it twice?
It might lead to overwriting something changed in the first one.

File: cspatients/util.py
Comment (KaitCrawford) [inline cspatients/util.py]: From what I can see, this method never actually saves the changes, it updates an instance of the PatientEntry object but with django you have to call patiententry.save() for the changes to actually be saved to the database. Otherwise they will just be lost as soon as the object is destroyed.

File: cspatients/util.py
Comment (KaitCrawford) [inline cspatients/util.py]: As above :) please move this closer to the `try`

File: cspatients/util.py
Comment (KaitCrawford) [inline cspatients/util.py]: You need to call `patiententry.save()` after changing this value, otherwise django will not write the change to the database.

File: cspatients/views.py
Comment (KaitCrawford) [inline cspatients/views.py]: Your view should really be the thing that handles your response to a request. I don't think delegating that to a util method is good practise.
If multiple views make use of this method then rather put it in the views.py file (I know this is contrary to what I told you when we discussed it but I think its the better option)
Or is this a channels thing?  since you seem to `print` the data in the method?

File: cspatients/util.py
Comment (KaitCrawford) [inline cspatients/util.py]: This method makes me feel uncomfortable. I'm sure either the Serializer or a form should handle this for you.

File: cspatients/views.py
Comment (KaitCrawford) [inline cspatients/views.py]: Take a look at https://docs.djangoproject.com/en/2.0/topics/http/shortcuts/#get-object-or-404
That method might help you remove some of the overhead you're doing in some places

**Review by KaitCrawford — COMMENTED** [review]
It's looking good. I've left a bunch of comments that I'd like to go through with you

**Comment by sewagodimo** [comment]
@nyashabryan even though I approved the PR don't merge before @KaitCrawford approved it

**Review by KaitCrawford — APPROVED** [review]
:+1: :cake: great job!

**Pull request merged**
